contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,365 @@
-      withdrawableAmountInNXM = totalPendingAmountInNXM;
-    }
-
-    return (totalPendingAmountInNXM, withdrawableAmountInNXM, withdrawableUntilIndex);
+    emit AssessmentDataForProductTypesSet(productTypeIds, cooldownPeriod, groupId);
   }
 
-  /* === MUTATIVE FUNCTIONS ==== */
-
-  /// Increases the sender's stake by the specified amount and transfers NXM to this contract
-  ///
-  /// @param amount  The amount of nxm to stake
-  function stake(uint96 amount) public whenNotPaused {
+  /* ========== MUTATIVE FUNCTIONS ========== */
 
-    stakeOf[msg.sender].amount += amount;
-    _tokenController().operatorTransfer(msg.sender, address(this), amount);
+  function addAssessorsToGroup(uint[] calldata assessorMemberIds, uint groupId) override external onlyContracts(C_GOVERNOR) {
+    // make new group id
+    if (groupId == 0) {
+      groupId = ++_groupCount;
+    }
 
-    emit StakeDeposited(msg.sender, amount);
+    uint length = assessorMemberIds.length;
+    for (uint i = 0; i < length; i++) {
+      uint assessorMemberId = assessorMemberIds[i];
+      require(assessorMemberId != 0, InvalidMemberId());
+      _groups[groupId].add(assessorMemberId);
+      _groupsForAssessor[assessorMemberId].add(groupId);
+      emit AssessorAddedToGroup(groupId, assessorMemberId);
+    }
   }
 
-  /// Withdraws a portion or all of the user's stake
-  ///
-  /// @dev At least stakeLockupPeriodInDays must have passed since the last vote.
-  ///
-  /// @param amount  The amount of nxm to unstake
-  /// @param to      The member address where the NXM is transfered to. Useful for switching
-  ///                membership during stake lockup period and thus allowing the user to withdraw
-  ///                their staked amount to the new address when possible.
-  function unstake(uint96 amount, address to) external override whenNotPaused {
-
-    uint stakeAmount = stakeOf[msg.sender].amount;
-    require(amount <= stakeAmount, InvalidAmount(stakeAmount));
+  function setGroupMetadata(uint groupId, bytes32 ipfsMetadata) override external onlyContracts(C_GOVERNOR) {
+    require(groupId > 0 && groupId <= _groupCount, InvalidGroupId());
 
-    uint govLockupExpiry = nxm.isLockedForMV(msg.sender);
-    require(block.timestamp >= govLockupExpiry || to == msg.sender, StakeLockedForGovernance(govLockupExpiry));
-
-    _unstake(msg.sender, amount, to);
+    _groupsMetadata[groupId] = ipfsMetadata;
+    emit GroupMetadataSet(groupId, ipfsMetadata);
   }
 
-  /// Withdraws all of the the given staker's stake
-  ///
-  /// @dev At least stakeLockupPeriodInDays must have passed since the last vote.
-  ///
-  /// @param staker  The address of the staker whose stake will be unstaked
-  function unstakeAllFor(address staker) external override whenNotPaused onlyTokenController {
-    _unstake(staker, stakeOf[staker].amount, staker);
+  function removeAssessorFromGroup(uint assessorMemberId, uint groupId) override external onlyContracts(C_GOVERNOR) {
```

Comment:
withdrawing membership should either automatically remove from groups, or we should prevent withdrawing membership while being in an assessment group.

Reply:
Tempting to proxy this through `TokenController` since `Registry` will already depend on TC and I don't really want to make Registry depend on `Assessment` as well

Reply:
I replaced `TC.removeFromWhiteList` with `TC.removeFromAssessorGroupsAndWhiteList`